src/backend/trace.rs

```diff
@@ -381,3 +433,240 @@ impl TraceData {
+                "NotSent" => ProbeStatus::NotSent,
+                "Awaited" => ProbeStatus::Awaited,
+                _ => ProbeStatus::Skipped,
+            }
+        }
+        fn host(&self) -> Option<IpAddr> {
+            self.host
+                .as_ref()
+                .map(|h| IpAddr::V4(Ipv4Addr::from_str(h).unwrap()))
+        }
+    }
+    impl From<ImportProbe> for Probe {
+        fn from(value: ImportProbe) -> Self {
+            Self {
+                sequence: value.sequence(),
+                identifier: value.identifier(),
+                src_port: value.src_port(),
+                dest_port: value.dest_port(),
+                ttl: value.ttl(),
+                round: value.round(),
+                sent: Some(value.sent()),
+                status: value.status(),
+                host: value.host(),
+                received: value.received(),
+                icmp_packet_type: Some(IcmpPacketType::NotApplicable),
+                extensions: None,
+            }
+        }
+    }
+    #[derive(Deserialize,Serialize,Debug,Clone)]
+    struct HopResults {
+        /// The total probes sent for this hop.
+        total_sent: usize,
+        /// The total probes received for this hop.
+        total_recv: usize,
+        /// The round trip time for this hop in the current round.
+        last: Option<Duration>,
+        /// The best round trip time for this hop across all rounds.
+        best: Option<Duration>,
+        /// The worst round trip time for this hop across all rounds.
+        worst: Option<Duration>,
+        /// The history of round trip times across the last N rounds.
+        mean: f64,
+        m2: f64,
+        /// The ABS(RTTx - RTTx-n)
+        jitter: Option<Duration>,
+        /// The Sequential jitter average calculated for each
+        javg: Option<f64>,
+        /// The worst jitter reading recorded
+        jmax: Option<Duration>,
+        /// The interval calculation i.e smooth
+        jinta: f64,
+    }
+    impl HopResults {
+        #[allow(clippy::too_many_arguments)]
+        fn new(
+            total_sent: usize,
+            total_recv: usize,
+            last: u64,
+            best: u64,
+            worst: u64,
+            mean: f64,
+            m2: f64,
+            jitter: u64,
+            javg: f64,
+            jmax: u64,
+            jinta: f64,
+        ) -> Self {
+                Self {total_sent,
+                total_recv,
+                last: Some(Duration::from_millis(last)),
+                best: Some(Duration::from_millis(best)),
+                worst: Some(Duration::from_millis(worst)),
+                mean,
+                m2,
+                jitter: Some(Duration::from_millis(jitter)),
+                javg: Some(javg),
+                jmax: Some(Duration::from_millis(jmax)),
+                jinta,
+            }
+        }
+        //TODO: Create combined struct ImportProbe Vec<ImportProbe> & HopResults.
```

Comment:
I think that is a good idea.  Can we use yaml rather than json as well?

Reply:
After looking at the TUN test code I had the same exact thought, make all test input files the same type.
Consider it done!